legend-engine-xt-relationalStore-trino-execution/src/main/java/org/finos/legend/engine/plan/execution/stores/relational/connection/ds/specifications/TrinoDatasourceSpecificationRuntime.java

```diff
@@ -0,0 +1,117 @@
+import org.finos.legend.engine.plan.execution.stores.relational.connection.authentication.strategy.DelegatedKerberosAuthenticationStrategy;
+import org.finos.legend.engine.plan.execution.stores.relational.connection.authentication.strategy.UserNamePasswordAuthenticationStrategy;
+import org.finos.legend.engine.plan.execution.stores.relational.connection.driver.DatabaseManager;
+import org.finos.legend.engine.plan.execution.stores.relational.connection.ds.specifications.keys.TrinoDatasourceSpecificationKey;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Properties;
+
+
+public class TrinoDatasourceSpecificationRuntime extends org.finos.legend.engine.plan.execution.stores.relational.connection.ds.DataSourceSpecification
+{
+
+    private static final String CLIENT_TAGS = "clientTags";
+    private static final String CATALOG = "catalog";
+    private static final String SSL = "SSL";
+    private static final String SSL_TRUST_STORE_PATH = "SSLTrustStorePath";
+    private static final String SSL_TRUST_STORE_PASSWORD = "SSLTrustStorePassword";
+    private static final String KERBEROES_REMOTE_SERVICE_NAME = "KerberosRemoteServiceName";
+    private static final String KERBEROS_USE_CANONICAL_HOSTNAME = "KerberosUseCanonicalHostname";
+
+    private static final String USER = "user";
+    private static final String PASSWORD = "password";
+    private final TrinoDatasourceSpecificationKey key;
+
+    public static final List<String> propertiesForDriver = Arrays.asList(CLIENT_TAGS, SSL, SSL_TRUST_STORE_PATH, SSL_TRUST_STORE_PASSWORD, KERBEROES_REMOTE_SERVICE_NAME, KERBEROS_USE_CANONICAL_HOSTNAME, USER, PASSWORD);
+
+    public TrinoDatasourceSpecificationRuntime(TrinoDatasourceSpecificationKey key, DatabaseManager databaseManager, AuthenticationStrategy authenticationStrategy, Properties extraUserProperties)
+    {
+        super(key, databaseManager, authenticationStrategy, extraUserProperties);
+        this.key = key;
+        this.extraDatasourceProperties.putAll(getProperties());
+    }
+
+    public TrinoDatasourceSpecificationRuntime(TrinoDatasourceSpecificationKey key, DatabaseManager databaseManager, AuthenticationStrategy authenticationStrategy, Properties extraUserProperties, int maxPoolSize, int minPoolSize)
+    {
+        super(key, databaseManager, authenticationStrategy, extraUserProperties, maxPoolSize, minPoolSize);
+        this.key = key;
+        this.extraDatasourceProperties.putAll(getProperties());
+    }
+
+    private Properties getProperties()
+    {
+        Properties properties = new Properties();
+
+        if (key.getClientTags() != null)
+        {
+            properties.setProperty(CLIENT_TAGS, key.getClientTags());
+        }
+
+        if (key.getCatalog() != null)
+        {
+            properties.setProperty(CATALOG, key.getCatalog());
+        }
+
+        AuthenticationStrategy authenticationStrategy = getAuthenticationStrategy();
```

Comment:
We should follow the pattern of the other spec classes. Delegate authn specific code to the AuthenticationStrategy classes instead of placing the logic here.

Reply:
Want to discuss further on this.